iohk-monitoring/src/Cardano/BM/Data/Tracer.lhs

```diff
@@ -354,7 +361,31 @@ trStructured verb tr = Tracer $ \arg ->
  in traceWith tracer =<< do
           meta <- mkLOMeta (getSeverityAnnotation arg) (getPrivacyAnnotation arg)
           return ( mempty
-                 , LogObject mempty meta (LogStructured obj)
+                 , LogObject mempty meta (LogStructuredText obj (T.pack $ show $ obj))
+                 )
+
+\end{code}
+
+
+\label{code:trStructuredText}\index{trStructuredText}
+\label{code:HasTextFormatter}\index{HasTextFormatter}
+\begin{code}
+class HasTextFormatter a where
+    formatText :: a -> Object -> Text
+    default formatText :: a -> Object -> Text
+    formatText _a = T.pack . show
+
+trStructuredText :: ( ToObject b, MonadIO m, HasTextFormatter b
+                    , HasPrivacyAnnotation b, HasSeverityAnnotation b )
+                 => TracingVerbosity -> Trace m a -> Tracer m b
+trStructuredText verb tr = Tracer $ \arg ->
```

Comment:
this tracer transformer requires traced types to have an instance of 'HasTextFormatter'